packages/core/src/browser/preferences/preference-provider.ts

```diff
@@ -58,22 +59,34 @@ export abstract class PreferenceProvider implements Disposable {
         this.toDispose.dispose();
     }
 
+    protected deferredChanges: PreferenceProviderDataChanges | undefined;
+
     /**
      * Informs the listeners that one or more preferences of this provider are changed.
      * The listeners are able to find what was changed from the emitted event.
      */
     protected emitPreferencesChangedEvent(changes: PreferenceProviderDataChanges | PreferenceProviderDataChange[]): void {
+        if (!this.deferredChanges) {
+            this.deferredChanges = {};
+        }
         if (Array.isArray(changes)) {
-            const prefChanges: PreferenceProviderDataChanges = {};
             for (const change of changes) {
-                prefChanges[change.preferenceName] = change;
+                this.deferredChanges[change.preferenceName] = change;
             }
-            this.onDidPreferencesChangedEmitter.fire(prefChanges);
         } else {
-            this.onDidPreferencesChangedEmitter.fire(changes);
+            this.deferredChanges = Object.assign(this.deferredChanges, changes);
         }
+        this.fireDidPreferencesChanged();
```

Comment:
use `setImmediate` instead to make sure that events are fired at the end of the same tick, not in next tick

Reply:
`setImmediate` is not standard API in the browser, so no